docs/terms/evaluation_parameter.md

```diff
@@ -1,15 +1,132 @@
+
+### Relationship to other objects
+
+Evaluation Parameters are used in Expectations when Validating data.  <TechnicalTag relative="../" tag="checkpoint" text="Checkpoints" /> use <TechnicalTag relative="../" tag="validation_action" text="Action" /> to store Evaluation Parameters in the <TechnicalTag relative="../" tag="evaluation_parameter_store" text="Evaluation Parameter Store" />.
+
+## Use cases
+
+<CreateHeader/>
+
+When creating expectations through the interactive method, Evaluation Parameters can be provided a temporary initial value.  This will allow you to test Expectations that are meant to rely on values from previous Validation runs before you have actually used them to Validate data.  Say you have created a Pandas dataframe called `my_df` that references a CSV that contains ten rows of data.  (To do so in the interactive process for creating Expectations, you would use the `great_expectations.read_csv()` method).  You want to create an expression that asserts that the row count for each Validation remains the same as the previous `upstream_row_count`, but since there is no previous `upstream_row_count` you need to provide a value that matches what the Expectation you are creating will find.
+
+To do so, you would define the parameter in question (`upstream_row_count`) by assigning it to the `$PARAMETER` value in a dictionary.  Then, you would provide the temporary value for that parameter by setting it as the value of the `$PARAMETER.<parameter_in_question>` key in the same dictionary.  Or, in this case, the `$PARAMETER.upstream_row_count`.
+
+For an example of this, see below:
+
+```python title="Python code"
+my_df.expect_table_row_count_to_equal(
+    value={"$PARAMETER": "upstream_row_count", "$PARAMETER.upstream_row_count": 10},
+    result_format={'result_format': 'BOOLEAN_ONLY'}
+)
+```
+
+This will return `{'success': True}`.
+
+An alternative method of defining the temporary value for an Evaluation Parameter is the `set_evaluation_parameter()` method, as shown below:
+
+```python title="Python code"
+my_df.set_evaluation_parameter("upstream_row_count", 10)
+```
+
+If the Evaluation Parameter's value is set in this way, you do not need to set it again (or define it alongside the use of the `$PARAMETER` key) for future Expectations.
+
+More typically, when validating Expectations, you can provide Evaluation Parameters that are only available at runtime.
+
+```python title="Python code"
+my_df.validate(
+    expectation_suite=my_dag_step_config, 
+    evaluation_parameters={"upstream_row_count": upstream_row_count}
+)
+```
```

Comment:
This approach avoids the use of URNs altogether -- is that intentional? That logic is currently buggy, but a patch is incoming. Otherwise, the previous doc on using eval params is essentially correct. An Evaluation Parameter can be defined by identifying a result in another suite:
```
suite_name = 'my_suite_1'
expectation_name = 'expect_table_row_count_to_equal'
metric = 'observed_value'
eval_param_urn = f'urn:great_expectations:validations:{suite_name}:{expectation_name}.result.{metric}'
validator.expect_table_row_count_to_equal(
   value={
      '$PARAMETER': eval_param_urn
   }
)
```

Reply:
I think I would recommend a second section of the Use Cases section to address this specific use case

Reply:
Added, along with `import great_expectations` for completeness.